src/core/net/ip6.hpp

```diff
@@ -133,6 +133,18 @@ class Ip6
     otError SendDatagram(Message &aMessage, MessageInfo &aMessageInfo, IpProto aIpProto);
 
     /**
+     * This method sends a IPv6 datagram as Ncp Host.
+     *
+     * @param[in]  aMessage          A reference to the message.
+     * @param[in]  aInterfaceId      The interface identifier of the network interface that received the message.
+     *
+     * @retval OT_ERROR_NONE   Successfully processed the message.
+     * @retval OT_ERROR_DROP   Message processing failed and the message should be dropped.
+     *
+     */
+    otError SendFromNcpHost(Message &aMessage, int8_t aInterfaceId);
```

Comment:
nit on the naming: this is not specific to NCP configurations.  This method is really closer in functionality to raw sockets.  How about calling this `SendRaw` and indicating in the doxygen that this method is for messages that already include a fully formed IPv6 header?